### PR TITLE
Refactor the fields keyword to 'record' for proper validation.

### PR DESCRIPTION
**Pull request by miguelTaningcoYoppworks**
### Issue

Currently, the Kalix-Study branch does not validate and therefore also fail in generating the static files. It turns out that the "fields" keyword has been changed such that projections require "records".

### Changes

- refactor 'fields' keywords to 'record foo'. eg. AllTenantIdsView will have the record AllTenantIdsFields

### Testing
- validated in local machine
- checked hugo server after translating

### CODE REVIEW

File: src/main/riddl/AllIdsProjections/allIdsProjections.riddl [hunk 1]
```diff
@@ -20,7 +20,7 @@ result AllSkus is {skus: SKU*}
 
 context AllIdsProjections is  {
   projection AllTenantIdsView is {
-     fields {
+     record fields {
```
Comment (JamesYopp) [inline src/main/riddl/AllIdsProjections/allIdsProjections.riddl]: I am curious, in most other areas of RIDDL you have a keyword, in this case record, followed by the name of that thing. 

So my question is, did the keyword change to record fields? or is it just record <name>?

In other words, if you changed this line to:
  record foo {
Would it still compile?

If yes, then should we be considering a different or better name for this record definition aside from fields?

Comment (reid-spencer) [reply src/main/riddl/AllIdsProjections/allIdsProjections.riddl]: Its just "record".  I changed the title of this PR :)

**Comment by reid-spencer** [comment]
@miguelTaningcoYoppworks - See the comments by James and me above. Your "Validate RIDDL" failed because of a syntax error. The run log also indicates, with this error:
```
Error:  Error: tenantProjections.riddl(9:5): Expected one of ("}" | "invariant" | "handler" | "include" | "term" | "{" | "fields" | "option" | "options" | "???"):
    record fields {
```
That the validator is using an old version of riddlc.  I will have to make a new release. This is a result of very new feature changes.  So we'll leave this sit until the new release can be configured.

**Comment by reid-spencer** [comment]
@miguelTaningcoYoppworks - I have released riddlc 0.18.1 which includes the projection parsing fixes and corrected the get-riddlc action to use it. The "validate-riddl" workflow should now use 0.18.1 and require the use of at least one "record" definition in a projection.  If you push the fix suggested by me in response to James' comment, it should all work! :)

File: src/main/riddl/AllIdsProjections/allIdsProjections.riddl (hunk 1 above)
Comment (miguelTaningcoYoppworks) [reply src/main/riddl/AllIdsProjections/allIdsProjections.riddl]: My bad for the vagueness. Thanks for updating the title! I've changed the 'fields' names to be more descriptive by having the View name and replacing View with Fields.

eg. AllTenantIdsView will have the record AllTenantIdsFields

**Review by JamesYopp — APPROVED** [review]
Looks good. Thanks Miguel!

**Pull request merged**